packages/fxa-auth-server/test/local/payments/configuration/manager.js

```diff
@@ -102,9 +102,24 @@ describe('PaymentConfigManager', () => {
     Container.set(AuthFirestore, firestore);
     Container.set(AuthLogger, {});
     Container.set(AppConfig, mockConfig);
+
     paymentConfigManager = new PaymentConfigManager();
     productConfigDbRef = paymentConfigManager.productConfigDbRef;
     planConfigDbRef = paymentConfigManager.planConfigDbRef;
+
+    // Ensure the collections are clean in case anything else might
+    // not have cleaned up properly. This helps reduce flaky tests.
+    await deleteCollection(
+      paymentConfigManager.firestore,
+      productConfigDbRef,
+      100
+    );
+    await deleteCollection(
+      paymentConfigManager.firestore,
+      planConfigDbRef,
+      100
+    );
+
```

Comment:
This and the retry below finally eliminated all flaky test behavior for me on this set of tests.